src/components/Courses/components/CourseCard/tests/courseCard.test.js

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { Provider } from "react-redux";
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import { mockedStoreCourses, mockedStoreUserAdmin, mockedStoreUserNotAdmin } from '../../../../../store/mockStore';
+import CourseCard from '../CourseCard';
+import formatDuration from '../../../../../helpers/pipeDuration';
+
+
+jest.mock('react-router-dom', () => ({
+  useNavigate: () => jest.fn(),
+}));
+
+const course = {
+  title: 'coursecard test title',
+  duration: 30,
+  creationDate: '9/3/2021',
+  description: 'description',
+  authors: [
+    "9b87e8b8-6ba5-40fc-a439-c4e30a373d36"
+  ]
+}
+const authors = {
+  name: 'author',
+  id: '9b87e8b8-6ba5-40fc-a439-c4e30a373d36',
+};
+
+test('CourseCard should display all course informations', () => {
+  render(
+    <Provider store={mockedStoreCourses}>
+      <CourseCard
+        key='0'
+        id='0'
+        title={course.title}
+        duration={course.duration}
+        creationDate={course.creationDate}
+        description={course.description}
+        authors={course.authors}
+      />
+    </Provider>
+  );
+  expect(screen.getByTestId('courseCardTitle').textContent).toEqual(course.title);
+  expect(screen.getByTestId('courseCardDescription').textContent).toEqual(course.description);
+  expect(screen.getByTestId('courseCardDuration').textContent).toEqual(`Duration:\xa0${formatDuration(course.duration)}\xa0hours`);
+  expect(screen.getByTestId('courseCardAuthors').textContent).toEqual(`Authors:\xa0${authors.name}`);
+  expect(screen.getByTestId('courseCardCreationDate').textContent).toEqual(`Created:\xa0${course.creationDate.replace(/\//g, '.')}`);
+});
+
+test('CourseCard should have additional buttons for admin user', () => {
+  render(
+    <Provider store={mockedStoreUserAdmin}>
+      <CourseCard
+        key='0'
+        id='0'
+        title={course.title}
+        duration={course.duration}
+        creationDate={course.creationDate}
+        description={course.description}
+        authors={course.authors}
+      />
+    </Provider>
+  );
+  expect(screen.getByTestId('courseCardUpdateButton')).toBeInTheDocument();
+  expect(screen.getByTestId('courseCardDeleteButton')).toBeInTheDocument();
+});
+
+test('CourseCard should not have additional buttons for not admin', () => {
+  render(
+    <Provider store={mockedStoreUserNotAdmin}>
+      <CourseCard
+        key='0'
+        id='0'
+        title={course.title}
+        duration={course.duration}
+        creationDate={course.creationDate}
+        description={course.description}
+        authors={course.authors}
+      />
+    </Provider>
+  );
+  expect(screen.queryByTestId('courseCardUpdateButton')).not.toBeInTheDocument();
+  expect(screen.queryByTestId('courseCardDeleteButton')).not.toBeInTheDocument();
```

Comment:
why do you make assertions twice here? same thing on lines 64-65

Reply:
not exactly the same. Here I am checking, for non admin user, if these buttons are hidden. And above, I am checking if these buttons are available for an admin user. I am providing different stores at these two tests

Reply:
Sorry, my question was specifically about duplication of lines 82-83

Reply:
On these lines, I am checking for 2 different buttons, `update` and `delete` buttons. Is that a duplication?

Reply:
Here I am checking if 2 buttons are missing, and in previous test I am checking if the same 2 buttons are present for different kind of user. These buttons are for `update` and `delete` actions. Is that wrong approach?

Reply:
My bad, didnt notice difference at first